src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -81,7 +77,12 @@ def _register_osrandom_engine(cls):
             result = cls.lib.ENGINE_set_RAND(engine, cls._osrandom_method)
             assert result == 1
             result = cls.lib.ENGINE_add(engine)
-            assert result == 1
+            if result != 1:
```

Comment:
No, it can probably fail for other reasons. I can assert that the error stack contains a specific code to confirm that. That's mostly safe (sometimes OpenSSL changes error codes, but it seems unlikely that they'd change this one).